### ISSUE TITLE
`install` via ssh reports non-existing permission issues

### ISSUE THREAD

**Issue by mih** [datalad/datalad#1184]
````
% datalad install -s git@github.com:datalad/datalad.git src/code
[INFO   ] Installing dataset at /tmp/some/src/code from git@github.com:datalad/datalad.git 
Permission denied (publickey).

1 installed item is available at
<Dataset path=/tmp/some/src/code>
````

but succeeds. For comparsion:

````
% git clone git@github.com:datalad/datalad.git
Cloning into 'datalad'...
remote: Counting objects: 28343, done.
remote: Total 28343 (delta 0), reused 0 (delta 0), pack-reused 28342
Receiving objects: 100% (28343/28343), 6.80 MiB | 1.22 MiB/s, done.
Resolving deltas: 100% (22018/22018), done.
````

No issue.

**Comment by mih** [comment]
Doesn't seem to be limited to `install`:

````
% datalad update -r
[INFO   ] Updating dataset 'data/src/raw' ... 
[INFO   ] Updating dataset 'data/src/tools' ... 
Permission denied (publickey).
````

but again exit code zero.

**Comment by bpoldrack** [comment]
re install: I get the same failure, but it happens for the plain git call too:

```
/tmp % git clone -v git@github.com:datalad/datalad.git /tmp/src/code
Cloning into '/tmp/src/code'...
Permission denied (publickey).
fatal: Could not read from remote repository.
```

**Comment by bpoldrack** [comment]
Okay, it's not a bug, it's a feature. If you use `git:` protocol SSH authentication is implied.
See https://help.github.com/articles/error-permission-denied-publickey/.

Just to be sure: Does it work for you via HTTPS, @mih ?

**Comment by mih** [comment]
This has been fixed in the meantime (tested in #1468). Closing.
